### PR TITLE
Reader tests

### PR DESCRIPTION
**Pull request by waywardmonkeys**
A long time ago, @cgay had done a quick skeleton for adding ``dfmc-reader`` tests. This takes that and then adds some initial tests that start to test reading the fragments for literals.

### CODE REVIEW

File: sources/dfmc/reader/tests/literal-test-suite.dylan [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+  assert-equal(f.fragment-value, '\\');
+  verify-presentation(f, "'\\\\'");
+
+  let f = read-fragment("'\\n'");
+  assert-equal(f.fragment-value, '\n');
+  verify-presentation(f, "'\\n'");
+
+  let f = read-fragment("'\\<01>'");
+  assert-equal(as(<integer>, f.fragment-value), 1);
+  verify-presentation(f, "'\\<1>'");
```
Comment (cgay) [inline sources/dfmc/reader/tests/literal-test-suite.dylan]: This seems to test basic \<...> syntax and handling of leading zeros.  How about a test to verify it's handling only hex digits?

File: sources/dfmc/reader/tests/literal-test-suite.dylan [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+end test character-literal-test;
+
+define test integer-literal-test ()
+  let f = read-fragment("123");
+  verify-literal(f, 123, <integer-fragment>);
+  verify-presentation(f, "123");
```
Comment (cgay) [inline sources/dfmc/reader/tests/literal-test-suite.dylan]: Add a test for negative numbers?

File: sources/dfmc/reader/tests/literal-test-suite.dylan [hunk 3]
```diff
@@ -0,0 +1,83 @@
...
+
+define test string-literal-test ()
+  let f = read-fragment("\"abc\"");
+  verify-literal(f, "abc", <string-fragment>);
+end test string-literal-test;
+
+define test symbol-literal-test ()
+  let kw = read-fragment("test:");
+  verify-literal(kw, #"test", <keyword-syntax-symbol-fragment>);
+  verify-presentation(kw, "test:");
```
Comment (cgay) [inline sources/dfmc/reader/tests/literal-test-suite.dylan]: How does it decide whether to present with keyword syntax or symbol syntax?

Comment (cgay) [inline sources/dfmc/reader/tests/literal-test-suite.dylan]: Nevermind, I see.

File: sources/dfmc/reader/tests/literal-test-suite.dylan [hunk 4]
```diff
@@ -0,0 +1,83 @@
...
+  verify-presentation(f, "123");
+end test integer-literal-test;
+
+define test ratio-literal-test ()
+  assert-signals(<ratios-not-supported>, read-fragment("1/2"));
+end test ratio-literal-test;
+
+define test string-literal-test ()
+  let f = read-fragment("\"abc\"");
+  verify-literal(f, "abc", <string-fragment>);
```
Comment (cgay) [inline sources/dfmc/reader/tests/literal-test-suite.dylan]: Add a test for "\"a\\nb\"" => "a\nb"?

**Comment by cgay** [comment]
LGTM.  I wish there were a "LGTM commit with or without addressing my comments" button.

**Comment by waywardmonkeys** [comment]
I'll add those tests that you mentioned before I merge this ... there are a lot of other tests to add over time as well. And this is just literals so far. :)
